lib/Echidna/ABI.hs

```diff
@@ -68,8 +71,9 @@ encodeSig n ts = n <> "(" <> T.intercalate "," (map abiTypeSolidity ts) <> ")"
 genSize :: MonadGen m => m Int
 genSize = (8 *) <$> Gen.enum 1 32
 
-genAbiAddress :: MonadGen m => m AbiValue
-genAbiAddress = let w64 = Gen.word64 $ constant minBound maxBound in
+genAbiAddress :: MonadGen m => Maybe [Addr] -> m AbiValue
```

Comment:
why not just move this into a `MonadReader` and we can avoid passing it a parameter at each use site?